doxygen/dox/sw_changes_1.12.dox

```diff
@@ -208,7 +208,7 @@ list the specific new feature that they were added for.
 \subsubsection subsubsec_rel_spec_112_change_0versus10_6_compat API Compatibility
 See API Compatibility Macros in HDF5 for details on using HDF5 version 1.12 with previous releases.
 
-[Compatibility report for Release 1.12.0 versus Release 1.10.6](Compatibility report for Release 1.12.0 versus Release 1.10.6)
+[Compatibility report for Release 1.12.0 versus Release 1.10.6]()
```

Comment:
The link for this is: https://htmlpreview.github.io/?https://github.com/HDFGroup/hdf5doc/blob/master/html/ADGuide/Compatibility_Report/hdf5-1.12-0-vs-hdf5-1.10.6-interface_compatibility_report.html

Reply:
Right - I plan a followup PR to address all the pre-1.14 reports - this change was just to fix the linkchecker temporarily.